homeassistant/components/light/lutron.py

```diff
@@ -0,0 +1,99 @@
+"""Support for Lutron lights."""
+import asyncio
+import logging
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light)
+from homeassistant.components.lutron import (
+    LutronDevice, LUTRON_DEVICES, LUTRON_GROUPS, LUTRON_CONTROLLER)
+
+DEPENDENCIES = ['lutron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
```

Comment:
The issue I was having is that in non-async call chain, when I used add_devices(devs, True), I end up in my callback calling schedule_update_ha_state with a partially initialized entity state with the stack below. self.hass in the entity isn't set, because we are calling update() in add_entities before scheduling async_add_entities has been called but after I have created my device and registered a callback with the hardware.

Exception in thread Thread-15:
Traceback (most recent call last):
 ...
  File "/home/pi/home-assistant/homeassistant/components/lutron.py", line 73, in _update_callback
    self.schedule_update_ha_state(True)
  File "/home/pi/home-assistant/homeassistant/helpers/entity.py", line 290, in schedule_update_ha_state
    self.hass.add_job(self.async_update_ha_state(force_refresh))
AttributeError: 'NoneType' object has no attribute 'add_job'